keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/artifacts/PublishedArtifact.kt

```diff
@@ -52,18 +52,18 @@ data class PublishedArtifact(
   val status: ArtifactStatus? = metadata["releaseStatus"]?.toString()
     ?.let { ArtifactStatus.valueOf(it) }
 
-  val createdAt = let {
-    metadata["createdAt"]
+  val createdAt: Instant?
+    get() = (metadata["createdAt"]
     //docker artifact createdAt time is under date field
-      ?: metadata["date"]
-  }
-    ?.let {
-      when (it) {
-        is Long -> Instant.ofEpochMilli(it) // to accommodate for artifact events from CI integration
-        is Instant -> it
-        else -> null
+      ?: metadata["date"])
+      ?.let {
+        when (it) {
+          is Long -> Instant.ofEpochMilli(it) // to accommodate for artifact events from CI integration
+          is Instant -> it
+          is String -> Instant.ofEpochMilli(it.toLong())
```

Comment:
You might want to wrap this in a try/catch in case there's a format issue in the string.

Reply:
done